internal/resource/configmap_test.go

```diff
@@ -168,6 +171,22 @@ var _ = Describe("GenerateServerConfigMap", func() {
 				Expect(configMapBuilder.Update(configMap)).To(Succeed())
 				Expect(configMap.Data).To(HaveKeyWithValue("userDefinedConfiguration.conf", expectedConfiguration))
 			})
+
+			When("user restricts SSL mechanisms to EXTERNAL", func() {
+				It("adds only EXTERNAL", func() {
+					userDefinedConfiguration := "auth_mechanisms.1 = EXTERNAL"
+					instance.Spec.Rabbitmq.AdditionalConfig = userDefinedConfiguration
+					expectedConfiguration := iniString(userDefinedConfiguration)
+
+					Expect(configMapBuilder.Update(configMap)).To(Succeed())
+					Expect(configMap.Data).To(HaveKeyWithValue("userDefinedConfiguration.conf", expectedConfiguration))
+					operatorDefaults, exists := configMap.Data["operatorDefaults.conf"]
+					Expect(exists).To(BeTrue())
+					Expect(operatorDefaults).NotTo(ContainSubstring("auth_mechanisms"))
+					Expect(operatorDefaults).NotTo(ContainSubstring("PLAIN"))
+					Expect(operatorDefaults).NotTo(ContainSubstring("ANONYMOUS"))
```

Comment:
This is fine, it works as expected. However, I would recommend the matcher [HaveKey()](https://onsi.github.io/gomega/#havekeykey-interface) instead of "expect bool to be true", because a failed assertion would give as little as "expected false to be true", whilst the other matcher would give a more helpful error.